python upgrade support: 3.12+

blangmt opened the pull request:
Summary: Support Python 3.11/3.12 in project torch

Test Plan: test changes in cpython_defs.c cpython_defs.h stub.c

Reviewed By: williamwen42

Differential Revision: D82250826




cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @kadeng @chauhang @amjames @Lucaskabela @jataylo @chenyang78

williamwen42 reviewed (changes requested):
Please try applying the changes in P1979911979

blangmt committed:
upgrade agios to py3.12 (#164597)

Summary:

Support Python 3.11/3.12 in project torch

Test Plan: test changes in cpython_defs.c cpython_defs.h stub.c

Reviewed By: williamwen42

Differential Revision: D82250826
